README.md

```diff
@@ -10,122 +10,274 @@
-  implementation, called `IMap`
-* additional data structures and simple messaging constructs such as
-  `Set`, `MultiMap`, `Queue`, `Topic`
-* cluster-wide unique ID generator, called `FlakeIdGenerator`
-* a distributed, [CRDT](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type)
-  based counter, called `PNCounter`
-* a cardinality estimator based on [`HyperLogLog`](https://en.wikipedia.org/wiki/HyperLogLog).
-
-Also, Hazelcast includes a production-ready
-[Raft](https://en.wikipedia.org/wiki/Raft_(computer_science))
-implementation which allows implementation of _linearizable_ constructs
-such as:
-
-* a distributed and reentrant lock implementation, called `FencedLock`
-* primitives for distributed computing such as `AtomicLong`,
-`AtomicReference` and `CountDownLatch`.
-
-Hazelcast data structures are in-memory, highly optimized and offer very
-low latencies. For a single `get` or `put` operation on an `IMap`, you
-can typically expect a round-trip-time of under _100 microseconds_.
-
-Additionally, Hazelcast provides a distributed batch and stream
-processing engine named Jet. It provides a Java API to build stream and
-batch processing applications through the use of a dataflow programming
-model. You can use it to process large volumes of real-time events or
-huge batches of static datasets. To give a sense of scale, a single node
-of Hazelcast has been proven to
-[aggregate 10 million events per second](https://jet-start.sh/blog/2020/08/05/gc-tuning-for-jet)
-with latency under 10 milliseconds.
+## What is Hazelcast
+
+Hazelcast is a distributed computation and storage platform for consistently
+low-latency querying, aggregation and stateful computation against event streams
+and traditional data sources. It allows you to quickly build resource-efficient,
+real-time applications. You can deploy it at any scale from small edge devices
+to a large cluster of cloud instances. 
+
+A cluster of Hazelcast nodes share both the data storage and computational load
+which can dynamically scale up and down. When you add new nodes to the cluster,
+the data is automatically rebalanced across the cluster. Running computational
+tasks (known as jobs) snapshot their state and scale with processing guarantees.
+
+## When to use Hazelcast
+
+Hazelcast provides a platform that can handle multiple types of workloads for
+building responsive applications.
+
+* Stateful data processing over streaming data or data at rest
+* Querying streaming and batch data sources directly using SQL
+* Ingesting data through a library of connectors and serving it using
+  low-latency SQL queries
+* Pushing updates to applications on events
+* Low-latency queue-based or pub-sub messaging  
+* Fast access to contextual and transactional data via caching patterns such as
+  read/write-through and write-behind)
+* Distributed coordination for microservices
+* Replicating data from one region to another or between data centers in the
```

Comment:
I know that we're trying to move away from the caching use case but it's still are major use case. I would suggest keeping it here somewhere at the end.

Reply:
I have " Fast access to contextual and transactional data via caching patterns such as
  read/write-through and write-behind)" - how do you think we can phrase it?

Reply:
Ah, missed that one :facepalm: LGTM